src/domains.rs

```diff
@@ -0,0 +1,199 @@
+//! Domain name related scanning, used by both email and URL scanners.
+//!
+//! This is called domains for familiarity but it's about the authority part of URLs as defined in
+//! https://datatracker.ietf.org/doc/html/rfc3986#section-3.2
+//!
+//! ```text
+//! authority   = [ userinfo "@" ] host [ ":" port ]
+//!
+//!
+//! userinfo    = *( unreserved / pct-encoded / sub-delims / ":" )
```

Comment:
Minor detail, but you never defined `pct-encoded` in this block while `unreserved` and `sub-delims` were defined.

Reply:
Thanks, I'll add that. Percent-encoding in the authority is not actually handled at the moment, but I'm not sure it's necessary. Have you seen any URLs using percent-encoding in the authority part in the wild? I think we can add it later if anyone requests it.